src/shell/output.c

```diff
@@ -94,9 +95,87 @@ static void shell_output_control (struct shell_output *out, bool stop)
     }
 }
 
+static int shell_output_data (struct shell_output *out, json_t *iodata)
+{
+    FILE *f;
+    const char *stream = NULL;
+    int rank;
+    char *data = NULL;
+    int len = 0;
+
+    if (iodecode (iodata, &stream, &rank, &data, &len, NULL) < 0) {
+        log_err ("iodecode");
+        return -1;
+    }
+    f = !strcmp (stream, "stdout") ? stdout : stderr;
+    if (len > 0) {
+        fprintf (f, "%d: ", rank);
+        fwrite (data, len, 1, f);
+    }
+    free (data);
+    return 0;
+}
+
+static void shell_output_commit_completion (flux_future_t *f, void *arg)
+{
+    struct shell_output *out = arg;
+
+    if (flux_future_get (f, NULL) < 0)
+        log_err ("shell_output_commit");
+    zlist_remove (out->pending_commits, f);
+    flux_future_destroy (f);
+}
+
+static int shell_output_commit (struct shell_output *out, json_t *entry)
```

Comment:
This code should probably not be limited to one eventlog entry per KVS commit, since that's just one extreme of what we will likely need to support.  I would suggest continuing to keep a list of entries in `struct shell_output`, and write this function so it aggregates all of them in one KVS transaction and empties the list.  For the initial implementation, we'll just call this function after every append, but later we can call it based on elapsed time and/or number of entries in the list.